c10/core/thread_pool.h

```diff
@@ -36,6 +36,10 @@ class C10_API TaskThreadPoolBase {
   virtual bool inThreadPool() const = 0;
 
   virtual ~TaskThreadPoolBase() noexcept {}
+
+  static size_t defaultNumThreads() {
+    return std::thread::hardware_concurrency();
```

Comment:
As far as I know this doesn't necessarily need to be consistent. It can return cores or threads in some environments.

Reply:
we could, for example, use `2 * get_nprocs()` on Linux systems and fallback to `std::thread::hardware_concurrency()` elsewhere

Reply:
though `2*` makes sense only on server.. we could either go with number of cores or just go with just simple `std::thread::hardware_concurrency()`

Reply:
After talking offline we decided that this will be resolved separately. The major constraint is that torch.get_num_threads returns the same number independent of the backend threadpool.